drain-service/drain_pretrained_inferencing.py

```diff
@@ -6,40 +6,84 @@
 
 # Third Party
 from opni_proto.log_anomaly_payload_pb import PayloadList
+from drain3.file_persistence import FilePersistence
 from drain3.template_miner import TemplateMiner
 from opni_nats import NatsWrapper
 
 logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(name)s - %(message)s")
-cp_template_miner = TemplateMiner()
+pretrained_template_miner = TemplateMiner()
 
 nw = NatsWrapper()
 
+def match_template(log_data, template_miner):
+    template, anomaly_level, cluster_id = template_miner.match(log_data.masked_log)
+    if template:
+        log_data.anomaly_level = anomaly_level
+        log_data.template_cluster_id = cluster_id
+        log_data.template_matched = template
+        log_data.inference_model = "drain"
+        return True
+    return False
+
+
 async def load_pretrain_model():
     # This function will load the pretrained DRAIN model for control plane logs in addition to the anomaly level for each template.
     try:
-        cp_template_miner.load_state("drain3_control_plane_model_v0.4.1.bin")
-        logging.info("Able to load the DRAIN control plane model with {} clusters.".format(cp_template_miner.drain.clusters_counter))
-        return True
+        pretrained_template_miner.load_state("drain3_control_plane_model_v0.6.0.bin")
```

Comment:
why is pertained_miner a global object(and the current is not)?

Reply:
So what happens is that the current_template_miner cluster_id index needs to know the number of clusters in the pretrained DRAIN model. So, the pretrained template miner can be a global object because you simply need to load the model but for the current template miner, you can't make that a global object because you need the number of clusters in the pretrained model.